src/megatron/energon/task_encoder/base.py

```diff
@@ -207,19 +338,51 @@ def _batch(
         else:
             raise ValueError("Unrecognized result type.")
 
+    def pre_pack(self, samples: List[T_sample]) -> List[List[T_sample]]:
+        """
+        For packing, selects the samples to be packed together. By default, no packing is happening.
+
+        Args:
+            samples: The samples to pre-pack. A full buffer will be passed into the function.
+
+        Returns: The pre-packed samples as a list of lists of samples.
+        """
+        raise NotImplementedError("Packing only effective when overridden.")
```

Comment:
I suggest a default identity implementation instead of raising?
```py
return [[s] for s in samples]
```

Reply:
I think we don't want to wrap the dataset in a PackingDataset if those methods are not defined by the user

Reply:
Yes, I suggest to keep the code below which will not use it. But still define the noop

Reply:
We don't want this method to be called ever. If this implementation is called, it's an error, hence we must raise it.